code/game/objects/effects/portals.dm

```diff
@@ -159,6 +159,10 @@ var/list/portal_cache = list()
 /obj/effect/portal/proc/teleport(atom/movable/M as mob|obj)
 	if(istype(M, /obj/effect)) //sparks don't teleport
 		return
+	if(istype(M, /atom/movable/light))
+		var/atom/movable/light/L = M
+		if (istype(L.holder, /obj/effect)) // sparks lights don't teleport either
+			return
```

Comment:
Should light overlays be teleported by portals *at all*? Legitimate question; I don't know the code well enough to know if that's actually intended.
Though I will say that if it is intended, it means that objects with lights will produce extra sparks and louder sounds when going through portals.

Reply:
They should, yes. It ensure that the flashlight still casts light when going through the portal.

Reply:
Why wouldn't whatever causes the light to move normally also cause it to move with the flashlight?

Reply:
... yeah, that's already handled by `follow_holder`. I will fix it.